### ISSUE TITLE
Unused parameter in shards.go [ DEL ]

### ISSUE THREAD

**Issue by yasharthdubey** [allegro/bigcache#399]
The key parameter in the `shards.go` is not being used here

```go
func (s *cacheShard) del(key string, hashedKey uint64) error {
	// Optimistic pre-check using only readlock
	s.lock.RLock()
	itemIndex := s.hashmap[hashedKey]

	if itemIndex == 0 {
		s.lock.RUnlock()
		s.delmiss()
		return ErrEntryNotFound
	}

	if err := s.entries.CheckGet(int(itemIndex)); err != nil {
		s.lock.RUnlock()
		s.delmiss()
		return err
	}
	s.lock.RUnlock()

	s.lock.Lock()
	{
		// After obtaining the writelock, we need to read the same again,
		// since the data delivered earlier may be stale now
		itemIndex = s.hashmap[hashedKey]

		if itemIndex == 0 {
			s.lock.Unlock()
			s.delmiss()
			return ErrEntryNotFound
		}

		wrappedEntry, err := s.entries.Get(int(itemIndex))
		if err != nil {
			s.lock.Unlock()
			s.delmiss()
			return err
		}

		delete(s.hashmap, hashedKey)
		s.onRemove(wrappedEntry, Deleted)
		resetKeyFromEntry(wrappedEntry)
	}
	s.lock.Unlock()

	s.delhit()
	return nil
}

**Comment by yasharthdubey** [comment]
can i pick it?

**Comment by janisz** [comment]
Yes please, if you like to go one step further you can setup golangci-lint action to find even more issues.

**Comment by yasharthdubey** [comment]
thanks

**Comment by yasharthdubey** [comment]
this is fixed in newer version was looking in v 1.0.1
